src/core/macro.scala

```diff
@@ -259,12 +281,223 @@ object Macro:
         .groupBy { case (name, _) => name }
         .toList
         .map { case (name, l) => name -> l.flatMap(_._2) }
-        .map { (name, anns) => Expr.ofTuple(Expr(name), Expr.ofList(anns)) }
+        .map { case (name, anns) => Expr.ofTuple(Expr(name), Expr.ofList(anns)) }
 
     private def isObjectOrScala(bc: Symbol) =
       bc.name.contains("java.lang.Object") || bc.fullName.startsWith("scala.")
 
     private def filterAnnotation(a: Term): Boolean =
       a.tpe.typeSymbol.maybeOwner.isNoSymbol ||
         a.tpe.typeSymbol.owner.fullName != "scala.annotation.internal"
-  }
+
+object MirrorlessMacro:
+  
+  object ValueClassDerivation:
+
+    inline def getValueClassParam[TC[_], V]: CaseClass.Param[TC, V] = 
+     ${ getValueClassParamImpl[TC, V] }
+
+    inline def rawContructByMacro[V](as: List[Any]): V = 
+      ${ rawContructByMacroImpl[V]('as) } 
+
+    def getValueClassParamImpl[TC[_]: Type, A: Type](using Quotes): Expr[CaseClass.Param[TC, A]] = 
+      import quotes.reflect.* 
+      
+      // repeated
+      def isObjectOrScala(bc: Symbol) =
+        bc.name.contains("java.lang.Object") || bc.fullName.startsWith("scala.")
+
+      def fromConstructor(from: Symbol): List[(String, List[Term])] = 
```

Comment:
shouldn't this be `annotationsFromConstructor`?

Reply:
No longer used but I have renamed the very same method in `CollectAnnotations` class.